v3/src/utilities/date-parser.ts

```diff
@@ -302,3 +302,12 @@ export function isDateString(iValue: any, iLoose?: boolean) {
     return spec.regex.test(iValue)
   }) || (!!iLoose && parseDateV3(iValue) != null)
 }
+
+// Regular expression to match ISO 8601 date strings as produced by Date.toISOString.
+// Note that this regular expression is more strict than the one used in parseDate (isoDateTimeRE) which supports
+// additional formats.
+const browserIsoDatePattern = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z$/
```

Comment:
Seems like we should support the [27-char](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString) format as well. Something like:
```suggestion
const browserIsoDatePattern = /^([+-]\d{6}|\d{4})-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z$/
```